src/colmap/estimators/cost_functions.h

```diff
@@ -301,6 +302,111 @@ class SampsonErrorCostFunction {
   const double y2_;
 };
 
+template <typename T>
+inline void EigenQuaternionToAngleAxis(const T* eigen_quaternion,
+                                       T* angle_axis) {
+  const T quaternion[4] = {eigen_quaternion[3],
```

Comment:
nit: Since you already map it to an Eigen::Quaternion, this could also be directly done using the Eigen object. Not sure if the implemented math is suitable for computing stable Jacobians with Jet though.